### DISCUSSION TITLE
Quite new to lamba / bref / aws, how does one develop locally

### DISCUSSION THREAD

**Discussion by iNilo** [brefphp/bref#1175] [Support]
I've got the gesture on how most of the stuff works.

I'm able to 
1) upload the code to aws (`serverless deploy`)
2) execute it with (`serverless invoke -f hello`)

But I'm struggeling with how to develop this script locally, without having to deploy it all the time.

How does one start developing locally and running the code locally? (so not having to deploy it every time)

I understood that the script returns a function () 
but this is all quite new for me, so how does one run my lamba function locally? <3 
(do I make a new php file that calls this file?)

Like how to I make `index.php` run locally?
```
<?php declare(strict_types=1);

require __DIR__ . '/vendor/autoload.php';

return function ($event) {
    return 'Hello ' . ($event['name'] ?? 'world');
};
```

Looking for any and all feedback, or just an example somewhere, or guidance

**Comment by iNilo** [comment]
Answering myself due some progress.

make a new file `run_local.php` 
```
<?php
(require('index.php'))(["event"]);
```

this will launch my function on my local machine, not remote

**Reply by mnapoli** [reply]
Yes that's a possibility, check out also the documentation: https://bref.sh/docs/function/local-development.html

**Reply by iNilo** [reply]
@mnapoli  I'm unsure how I missed that 🤷 🤦

**Reply by mnapoli** [reply]
haha no problem!
